testcode.c

```diff
@@ -0,0 +1,87 @@
+#include <assert.h>
+#include "CurrentMeasurementRange.h"
+
+int main()
+{
+int *ptr;
+int rangeinfo[100][4];
+  
+printf("Test case 1");
+int samplesarray[] = {4,5};
+int arraysize = sizeof(samplesarray) / sizeof(samplesarray[0]);
+ptr = detectAndPrintRangeDetails(samplesarray,arraysize);
+//Test case to check start of range
+assert(*(ptr+4) == 4);
+//Test case to check end of range
+assert(*(ptr+5) == 5);
+//Test case to check number of readings in the range
+assert(*(ptr+6) == 2);
+//Test case to check number of range detected
+assert(*(ptr+7) == 1);  
+  
+printf("\n\nTest case 2");
+int samplesarray1[] = {3, 3, 5, 4, 10, 11, 12};
+int arraysize1 = sizeof(samplesarray1) / sizeof(samplesarray1[0]);
+ptr = detectAndPrintRangeDetails(samplesarray1,arraysize1);
+//Test case to check start of range
+assert(*(ptr+4) == 3);
```

Comment:
The above comment will make the test also more readable.
Treat the tests as 'example code' which you can give to the clients of your library. If this is tough to explain, the chance that they will make mistakes is more.